server/sonar-server/src/main/java/org/sonar/server/settings/ws/ValuesAction.java

```diff
@@ -211,6 +207,14 @@ private void processSettings() {
       return valueBuilder;
     }
 
+    private void setInherited(Setting setting, Settings.Setting.Builder valueBuilder) {
+      boolean isDefault = setting.isDefault();
+      boolean isGlobal = !component.isPresent();
+      boolean isOnComponent = component.isPresent() && Objects.equals(setting.getComponentId(), component.get().getId());
```

Comment:
component is the component from the request, it could be renamed ``requestedComponet``